templates/task/cf.yml

```
@@ -0,0 +1,105 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+AWSTemplateFormatVersion: "2010-09-09"
+Description: "CloudFormation template that represents a task on Amazon ECS."
+Parameters:
+  TaskName:
+    Type: String
+  TaskCPU:
+    Type: String
+  TaskMemory:
+    Type: String
+  LogRetention:
+    Type: Number
+    Default: 30
+  ContainerImage:
+    Type: String
+  TaskRole:
+    Type: String
+  Command:
+    Type: String
+Conditions:
+  HasImage:
+    !Not [!Equals [!Ref ContainerImage, ""]]
+  UseDefaultTaskRole:
+    !Equals [!Ref TaskRole, ""]
+  HasCommand:
+    !Not [!Equals [!Ref Command, ""]]
+Resources:
+  TaskDefinition:
+    Condition: HasImage
+    Type: AWS::ECS::TaskDefinition
+    DependsOn: ECRRepo
+    Properties:
+      ContainerDefinitions:
+        -
+          Image: !Ref ContainerImage
+          Command: !If [HasCommand, !Split [" ", !Ref Command], !Ref "AWS::NoValue"]
+          LogConfiguration:
+            LogDriver: awslogs
+            Options:
+              awslogs-region: !Ref AWS::Region
+              awslogs-group: !Ref LogGroup
+              awslogs-stream-prefix: copilot-task
+          Name: !Ref TaskName
+          {{if .EnvVars}}Environment:{{range $name, $value := .EnvVars}}
+          - Name: {{$name}}
+          - Value: {{$value}}{{end}}{{end}}
```

to prevent generating extra new line if there's no env vars.
```suggestion
          Name: !Ref TaskName{{if .EnvVars}}
          Environment:{{range $name, $value := .EnvVars}}
          - Name: {{$name}}
          - Value: {{$value}}{{end}}{{end}}
```

---

oh no i thought I moved it to the same line as the `Name` property...!